session.go

```diff
@@ -234,13 +239,30 @@ func (s *session) finishTxn(rollback bool) error {
 			s.txn.SetOption(kv.BinlogData, bin)
 		}
 	}
-	err := s.txn.Commit()
-	if err != nil {
+
+	if err := s.checkSchemaValid(); err != nil {
+		if !s.sessionVars.RetryInfo.Retrying {
+			err = s.Retry()
```

Comment:
`domain.ErrInfoSchemaExpired` is not `kv.IsRetryableError`, `Retry` will only retry once.